### PR TITLE
Med: Build: Update gnutls build requirements.

### PR DESCRIPTION
**Pull request by clumens**
The macros needed to enable/disable FIPS mode were not introduced until 3.6.2.  Pacemaker supports building on systems as late as Ubuntu 18, which has 3.5.18.

So, make 3.5.18 the required version and add a note that we really do prefer 3.6.2.  Then, add a configure test and compability macros to handle the gap between the minimum required version and the preferred version.

This means that anyone with gnutls >= 3.5.18 and < 3.6.2 cannot run Pacemaker in FIPS mode.  In practice, this should not be a problem because it only affects the following people:

* Ubuntu 18 users - The commits that switched us to using gnutls were only on the main (Pacemaker 3) branch, and we do not expect that Ubuntu will backport Pacemaker 3 to that release.

* People building from source - No one that cares about FIPS mode will be building from source.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -18,12 +18,14 @@
...
 | 0.18 or later   |                    |                    | autopoint      |
-| 2.12.0 or later | gnutls-devel       | libgnutls-devel    | libgnutls-dev  |
+| 3.5.18 or later | gnutls-devel       | libgnutls-devel    | libgnutls-dev  |
 
 Also:
 * make must be GNU (or compatible) (setting MAKE=gmake might also work but is
   untested)
 * GNU (or compatible) getopt must be somewhere on the PATH
+* gnutls >= 3.6.2 is preferred.  Pacemaker can be built with versions back to
+  3.5.18 but will not be useable in FIPS mode.
```
Comment (kgaillot) [inline INSTALL.md:28]: Actually 2.12.0 or later is still accurate for the minimum. Versions 3.3.0 to 3.6.1 will not be usable in FIPS mode.

File: configure.ac [hunk 2]
```diff
@@ -1643,10 +1643,14 @@ dnl ========================================================================
...
 dnl ========================================================================
 
-PKG_CHECK_MODULES(GNUTLS, [gnutls >= 2.12.0],
+PKG_CHECK_MODULES(GNUTLS, [gnutls >= 3.5.18],
                   [CPPFLAGS="${CPPFLAGS} ${GNUTLS_CFLAGS}"
                    LIBS="${LIBS} ${GNUTLS_LIBS}"])
 
+AC_CHECK_FUNCS([gnutls_fips140_set_mode])
+AS_IF([test x"$ac_cv_func_gnutls_fips140_set_mode" = x"yes"],
+      AC_DEFINE(HAVE_FIPS_SET_MODE, 1, [Define this if gnutls_fips140_set_mode exists]))
```
Comment (kgaillot) [inline configure.ac:1652]: `AC_CHECK_FUNCS()` should automatically define `HAVE_GNUTLS_FIPS140_SET_MODE`

**Review by kgaillot — COMMENTED** [review]
looking good

**Review by kgaillot — APPROVED** [review]
